File: Technik/search_algorithm.py
# This is the list with all possible values
# put all your trash in here:
lst: list = [
    "simon",
    "niklas",
    "steven",
    "phi-anh"
]

def search(values: list, query: str) -> [bool, int]:
    # This is the string-sequence the algorithm should search for in the list. 
    # Should be in lower-case... MUST be in lower-case!
    query = query.lower()

    # Index of the query. This will be changed by the for-loop.
    # Better don't have or you will break everything like always!
    idx: int = 0

    # Iterate through every object in list
    for value in values:
        
        # is q in lower(entry) ?
        # we have to make the value lower, because we want to check for "A" in "A" AND "a":
            # entry = "sImOn"
            # entry         == "simon": false
            #  entry.lower() == "simon": true
        if query in value.lower():
            # return True and found index
            return True, idx
        else:
            # count index 1 up
            idx += 1

    # if value was not found in list, return False and -1.
    return False, -1
    
# search list "lst" for "phi-anhh"
found, index = search(lst, "phi-anh")

# if it was found, print "Gefunden an Index :index:"
if found:
    print(f"Gefunden an Index {index}")

# otherwise "Leider nicht gefunden. (res: :index:)"
else:
    print(f"Leider nicht gefunden. (res: {index})")

File: Technik/SearchAlgorithm.java
package Technik;

public class SearchAlgorithm {

    public static final int NO_KEY = -1;

    public static void main(final String[] args) {

        // This is the string-sequence the algorithm should search for in the list. 
        final String query = "simon";

        // This is the list with all possible values put all your trash in here:
        final String[] values = { "niklas", "simon", "steven", "lasse", "phi-anh" };

        final int result = findInArray(values, query);

        // if result is -1 (NO_KEY) return "Nicht gefunden"
        if (result == SearchAlgorithm.NO_KEY) {
            System.out.println("Nicht gefunden");
        } else {
        // otherwise "Gefunden"
            System.out.println("Gefunden an Index: " + result);
        }
    }

    /**
     * Searches a specific string in an string array
     * 
     * @param values    The values to look for the query
     * @param query     The string to look for in the values
     * 
     * @return Index of value in array
     */
    private static int findInArray(final String[] values, final String query) {
        for (int i = 0; i < values.length; i++) {
            final String value = values[i];
            if (value.equalsIgnoreCase(query)) {
                return i;
            }
        }

        return SearchAlgorithm.NO_KEY;
    }

}